### ISSUE TITLE
piecewise BUG: all the input arrays must have same number of dimensions

### ISSUE THREAD

**Issue by vlachoudis** [numpy/numpy#6966]
The following bug I first saw it from 1.9.2+
While clearly was working on 1.8.2 and before.

`import numpy as np`
`D=np.asarray([[0.,1],[2.,3.]])`
`np.piecewise(D,[D<=1, D>2],[0,1,0.5])`
**_ValueError: all the input arrays must have same number of dimensions**_

Which according to the manual the condlist can be one less from the funclist
If I remove the last argument in the funclist it works.

**Comment by bsipocz** [comment]
Please close, this now works again (I haven't checked what fixed it, but it didn't work in 1.10.4).

```
In [40]: import numpy as np 
    ...: D=np.asarray([[0.,1],[2.,3.]]) 
    ...: np.piecewise(D,[D<=1, D>2],[0,1,0.5])                                                                          
Out[40]: 
array([[0. , 0. ],
       [0.5, 1. ]])
In [41]: np.__version__                                                                                                 
Out[41]: '1.18.0.dev0+67ecbcf'
```

**Comment by mattip** [comment]
Closing. Please reopen with more details if the issue is still relevant.
